src/api/link_post.js

```diff
@@ -101,10 +163,42 @@ class LinkPostHandler {
       cb({ code: 400, message: 'no valid did in the consent_msg' })
       return
     }
-
     const did = didMatch[0]
 
-    return {msg, sig, did}
+    if (!!chainId && !isSupportedChainId(chainId)) {
+      cb({ code: 400, message: 'unsupported chainId provided' })
+      return
+    }
+
+    if (!!type && !isValidAccountType(type)) {
+      cb({ code: 400, message: 'unsupported type provided' })
+      return
+    }
+
+    if (type === ACCOUNT_TYPES.erc1271 && !address) {
+      cb({ code: 403, message: 'no address' })
```

Comment:
Maybe make this message a bit clearer. Something like `Parameter "address" needed for type erc1271`.

Reply:
👍